### PR TITLE
Update NNCF version to 1.5

### PR DESCRIPTION
**Pull request by kshpv**

### CODE REVIEW

**Comment by kshpv** [comment]
run tests

**Comment by LeonidBeynenson** [comment]
2 @aleksei-kashapov :    
Aleksei, please, 
* uncomment `prepare_for_export` in `tools/export.py`
* grep `1.4.1` in documentation and replace it with `1.5.0`

File: requirements/nncf_compression.txt [hunk 1]
```diff
@@ -1 +1 @@
-nncf==1.4.1
+nncf==1.5
```
Comment (LeonidBeynenson) [inline requirements/nncf_compression.txt]: `1.5.0`

File: mmdet/integration/nncf/compression.py [hunk 2]
```diff
@@ -144,11 +144,33 @@ def dummy_forward(model):
...
+            return args, kwargs
+
+        # during building original graph
+        if not kwargs.get('return_loss') and kwargs.get('forward_export'):
+            return args, kwargs
+
+        # during model's forward
+        assert 'img' in kwargs
+        img = kwargs['img']
+        if isinstance(img, list):
```
Comment (LeonidBeynenson) [inline mmdet/integration/nncf/compression.py:160]: Please, add `assert len(img) == 1`

Also, please, add explanation messages for all `assert`-s in this function

**Comment by kshpv** [comment]
run tests

**Comment by LeonidBeynenson** [comment]
2 @Ilya-Krylov, @aleksei-kashapov :   
I'm generally accept this change, but I ask Aleksei to check full pipeline on one more model (e.g. `retinanet`).

Aleksei, please, write when you check it, and IMHO this PR will be ready for merge.

**Comment by kshpv** [comment]
I've done the experiment that Leonid asked. `retinanet `model works fine for all possible scenario (train, test, export, test_exported)

**Comment by LeonidBeynenson** [comment]
2 @Ilya-Krylov:    
Ilya, I think this PR is fine. Please, merge it if you agree with it.

**Comment by Ilya-Krylov** [comment]
run tests

**Review by druzhkov-paul — CHANGES_REQUESTED** [review]
@aleksei-kashapov Please add NNCF version to `mmdet/utils/collect_env.py` and checkpoints meta.

**Comment by kshpv** [comment]
> @aleksei-kashapov Please add NNCF version to `mmdet/utils/collect_env.py` and checkpoints meta.

@druzhkov-paul `mmdet/utils/collect_env.py` - Done
could you specify where I should add changes to add NNCF version to checkpoints meta?

**Comment by druzhkov-paul** [comment]
Ok, I see that it's already in there. Thanks.

**Comment by druzhkov-paul** [comment]
run tests

**Review by druzhkov-paul — APPROVED** [review]

**Comment by Ilya-Krylov** [comment]
This branch is out-of-date with the base branch

**Comment by kshpv** [comment]
run tests

**Pull request merged**